src/lib/atom-private.c

```diff
@@ -1025,6 +1029,33 @@ _lldpctl_atom_get_str_port(lldpctl_atom_t *atom, lldpctl_key_t key)
 	}
 }
 
+static lldpctl_atom_t*
+_lldpctl_atom_set_str_port(lldpctl_atom_t *atom, lldpctl_key_t key,
+    const char *value)
+{
+	struct _lldpctl_atom_port_t *p =
+	    (struct _lldpctl_atom_port_t *)atom;
+	struct lldpd_port     *port     = p->port;
+
+	if (!value || !strlen(value))
+		return NULL;
+
+	switch (key) {
+	case lldpctl_k_port_id:
+		if (port->p_id != NULL) {
+			free(port->p_id);
```

Comment:
`free` is already checking if the value is `NULL`, you don't need to check again.